sph/include/sph/eos.hpp

```diff
@@ -18,33 +18,77 @@ namespace sph
  *
  * In this simple version of state equation, we calculate all depended quantities
  * also for halos, not just assigned particles in [startIndex:endIndex], so that
- * we can avoid halo exchange of p and c.
+ * we could potentially avoid halo exchange of p and c in return for exchanging halos of u.
  */
 template<typename Dataset>
 void computeEquationOfState(size_t /*startIndex*/, size_t /*endIndex*/, Dataset& d)
 {
-    using T = typename Dataset::RealType;
+    using T = std::decay_t<decltype(d.u[0])>;
 
     // const T R     = 8.317e7;
     const T gamma = (5.0 / 3.0);
 
-    const T* rho = d.rho.data();
-    const T* u   = d.u.data();
-    // const T* mui = d.mui.data();
+    const auto* u  = d.u.data();
+    const auto* kx = d.kx.data();
+    const auto* xm = d.xm.data();
+    const auto* m  = d.m.data();
+    // const auto* mui = d.mui.data();
 
     const size_t numParticles = d.x.size();
 
-    // T* cv   = d.cv.data();
-    // T* temp = d.temp.data();
-    T* p = d.p.data();
-    T* c = d.c.data();
+    // auto* cv   = d.cv.data();
+    // auto* temp = d.temp.data();
+    auto* p = d.p.data();
+    auto* c = d.c.data();
 
 #pragma omp parallel for schedule(static)
     for (size_t i = 0; i < numParticles; ++i)
     {
         // cv[i]   = T(1.5) * R / mui[i];
         // temp[i] = u[i] / cv[i];
-        T tmp = u[i] * (gamma - 1);
+        T tmp = u[i] * (gamma - T(1));
+        T rho = kx[i] * m[i] / xm[i];
+        p[i]  = rho * tmp;
+        c[i]  = std::sqrt(tmp);
+    }
+}
+
+/*! @brief Ideal gas EOS for SPH formulations where rho is stored
+ *
```

Comment:
Is this version saved temporarily for checking?

Reply:
yes